sar-qsar-environ-res.csl

```diff
@@ -0,0 +1,164 @@
+  <info>
+    <title>SAR and QSAR in Environmental Research</title>
+    <title-short>SAR QSAR Environ Res</title-short>
+    <id>http://www.zotero.org/styles/sar-qsar-environ-res</id>
+    <link href="http://www.zotero.org/styles/sar-qsar-environ-res" rel="self"/>
+    <link href="http://www.zotero.org/styles/elsevier-without-titles" rel="template"/>
+    <link href="http://www.tandf.co.uk/journals/authors/style/reference/ref_gsar.pdf" rel="documentation"/>
+    <author>
+      <name>David Bajusz</name>
+    </author>
+    <contributor>
+      <name>Richard Karnesky</name>
+      <email>karnesky+zotero@gmail.com</email>
+      <uri>http://arc.nucapt.northwestern.edu/Richard_Karnesky</uri>
+    </contributor>
+    <contributor>
+      <name>Rintze Zelle</name>
+      <uri>http://twitter.com/rintzezelle</uri>
+    </contributor>
+    <category citation-format="numeric"/>
+    <category field="generic-base"/>
+    <summary>A style for SAR and QSAR in Environmental Science and possibly other journals of Taylor and Francis.</summary>
+    <updated>2015-06-01T15:26:07+00:00</updated>
+    <rights license="http://creativecommons.org/licenses/by-sa/3.0/">This work is licensed under a Creative Commons Attribution-ShareAlike 3.0 License</rights>
+  </info>
+  <macro name="author">
+    <names variable="author" suffix=", ">
+      <name and="text" delimiter-precedes-et-al="never" delimiter-precedes-last="never" et-al-min="0" initialize-with="."/>
+      <label form="short" prefix=", "/>
+      <substitute>
+        <names variable="editor"/>
+        <names variable="translator"/>
+      </substitute>
+    </names>
+  </macro>
+  <macro name="editor">
+    <names variable="editor">
+      <name and="text" delimiter-precedes-last="never" initialize-with="."/>
+      <label form="short" text-case="lowercase" prefix=", "/>
+    </names>
+  </macro>
+  <citation collapse="citation-number">
+    <sort>
+      <key variable="citation-number"/>
+    </sort>
+    <layout prefix="[" suffix="]" delimiter=",">
+      <text variable="citation-number"/>
+    </layout>
+  </citation>
+  <bibliography entry-spacing="0" second-field-align="flush" et-al-min="7" et-al-use-first="6">
+    <layout suffix=".">
+      <text variable="citation-number" prefix="[" suffix="]"/>
+      <choose>
+        <if type="article-journal" match="any">
+          <text macro="author"/>
+          <text variable="title" font-style="italic" suffix=", "/>
+          <text variable="container-title" form="short" suffix=" "/>
+          <number suffix=" " variable="volume"/>
+          <date date-parts="year" form="text" variable="issued" prefix="(" suffix="), "/>
+          <text variable="page" prefix="pp. "/>
+        </if>
+        <else-if type="book" match="any">
+          <text macro="author"/>
+          <text variable="title" font-style="italic" suffix=", "/>
+          <choose>
+            <if match="any" variable="edition">
```

Comment:
use a more robust solution here. Copy the edition macro from a style like Elsvier Vancouver:
https://github.com/citation-style-language/styles/blob/master/elsevier-vancouver.csl#L52
and call it here. 
No need to test for variable before using it.